### ISSUE TITLE
Adopt GitFlow

### ISSUE THREAD

**Issue by HEdingfield** [BrightSpots/rcv#289]
From now on, I strongly recommend we adopt [the GitFlow workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) for this project. This is driven by the fact that we're now starting to use official version numbers, so we have to be really careful about our release process.

I went ahead and created the `develop` branch, which we should now treat as our primary branch for merging other `feat-*` branches into. In IntelliJ, I recommend starring both the remote and your local `develop` branches in the workspace menu in the bottom right corner.

I'll assign myself as the whipcracker here, and I'll close this issue once we've used GitFlow to officially release 1.0.1 or 1.1.0 just to make sure we're all clear on how it works.

It's definitely worth reading the whole post on GitFlow, but here's a summary from the bottom for a quick reminder:

1. A `develop` branch is created from `master`
2. A `release` branch is created from `develop`
3. `Feature` branches are created from `develop`
4. When a `feature` is complete it is merged into the `develop` branch
5. When the `release` branch is done it is merged into `develop` and `master`
6. If an issue in `master` is detected a `hotfix` branch is created from `master`
7. Once the `hotfix` is complete it is merged to both `develop` and `master`

**Comment by HEdingfield** [comment]
The main thing to remember is to base your PRs on the `develop` branch, not `master`. In fact, I'd recommend deleting your local copy of `master` entirely.

I set up a couple branch rules [here](https://github.com/BrightSpots/rcv/settings/branches) to help out with this. Basically merges into `develop` now require an approval from at least 1 reviewer, and merges into `master` require approvals from at least 2 reviewers (I thought this was reasonable, since pushes to this branch will be relatively rare, and we want to be extra careful with them). We can change these rules or add others too, just let me know.

**Comment by HEdingfield** [comment]
Release process, for future reference:

- Merged the PR from `develop` to `master`
- Switched to `master` on my local machine and made sure all tests passed
- Ran `jlinkZip` in the Gradle menu and renamed the resulting .zip file "universal_rcv_tabulator_v[version]_[os].zip" (e.g. "universal_rcv_tabulator_v0.1.0-beta_windows.zip")
- Unzipped locally and tested GUI and CLI using all provided sample data files
- Followed [these instructions](https://help.github.com/en/articles/creating-releases), uploading the zip as part of the release
- Published officially, pulled `master` again locally, verified new version tag was showing up in the commit log, deleted local `master` branch

**Comment by HEdingfield** [comment]
Transferred this info to the [Wiki](https://github.com/BrightSpots/rcv/wiki).
